File: mobile/app/src/main/java/com/bellako/kiwi/userSettings/types/UserSettingsDTO.kt
package com.bellako.kiwi.userSettings.types

import com.bellako.kiwi.users.Email

data class UserSettingsDTO(
    val email: String,
    val areNotificationsEnabled: Boolean,
    val theme: Theme

) {
    fun toState() : UserSettingsState {
        return UserSettingsState(
            email = email,
            areNotificationsEnabled = areNotificationsEnabled,
            theme = theme,
        )
    }

    fun toDomainObject(): Result<UserSettings> {
        return Email.of(email).map { validEmail ->
            UserSettings(
                email = validEmail,
                areNotificationsEnabled = areNotificationsEnabled,
                theme = Theme.valueOf(theme.name)
            )
        }
    }
}

File: frontend/src/userSettings/types/UserSettingsDTO.ts
import {ThemeOption, UserSettings} from "./UserSettings";
import {Email} from "../../users/Email";

export interface UserSettingsDTO {
    email: string;
    areNotificationsEnabled: boolean;
    theme: ThemeOption;
}

export const toDomainObject = (dto: UserSettingsDTO): UserSettings => ({
    email: Email.of(dto.email),
    areNotificationsEnabled: dto.areNotificationsEnabled,
    theme: dto.theme,
});